Automattic/woocommerce-payments
# RPP - Add phone number check

Fixes #7412

#### Changes proposed in this Pull Request

- Add the check of phone number before processing anything else in `InitialState`. 
- Add tests to cover 100% for new code. 









#### Testing instructions





* Try to check out an order.
* In the billing phone, choose this number `242342342324234234232423423423` or any number with length > 20. 
* See this error message in the checkout page: 

![Markup on 2023-10-25 at 22:59:07](https://github.com/Automattic/woocommerce-payments/assets/10045087/ccfc4744-4ea0-401e-a1da-ab2f315c2afd)

-------------------


**Post merge**
